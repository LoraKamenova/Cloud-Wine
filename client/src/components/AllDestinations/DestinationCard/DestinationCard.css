.destination-card {
    display: flex;
    align-items: center;
    background-color: white;
    width: 100%;
    height: 170px ;
    padding: 20px;
    border-radius: 100px;
    margin: 10px;
    box-shadow: 0px 0px 11px -4px #000000;
}

.destination-card-image {
    margin-right: 50px;
    border-radius: 50%;
    width: 150px;
    height: 150px;
}

.destination-card-info-wrapper {
    display: flex;
    flex-direction: column;
}

.destination-info-item {
    font-size: 16px;
}

.destination-card-line-wrapper {
    display: flex;
    margin: 5px 0;
}

.destination-card-pic {
    width: 25px;
    height: 25px;
    margin-right: 10px;
}
