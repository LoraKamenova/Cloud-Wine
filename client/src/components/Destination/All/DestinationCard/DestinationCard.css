.destination-card {
    display: flex;
    align-items: center;
    width: 100%;
    height: 170px ;
    background-color: white;
    padding: 20px;
    margin: 10px;
    border-radius: 100px;
    box-shadow: 0px 0px 11px -4px #000000;
}

.destination-card-image {
    width: 130px;
    height: 130px;
    margin-right: 35px;
    margin-left: 35px;
}

.destination-card-info-wrapper {
    display: flex;
    flex-direction: column;
}

.destination-card-line-wrapper {
    display: flex;
    margin: 5px 0;
}

.destination-card-pic {
    width: 25px;
    height: 25px;
    margin-right: 10px;
}

.destination-info-item {
    font-size: 16px;
}
