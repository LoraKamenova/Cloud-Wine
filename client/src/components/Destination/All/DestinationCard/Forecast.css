/*.forecast-section {*/
/*    width: 95%;*/
/*}*/

.forecast-wrapper {
    color: white;
    width: 90%;
    height: 100%;
    display: flex;
    flex-direction: column;
    /*justify-content: center;*/
    background-color: #FFFACC;
    border-radius: 5px;
    padding: 20px 30px 70px 30px;
    margin: auto;
    background-image: url("https://media.istockphoto.com/photos/clear-blue-sky-background-with-scattered-clouds-picture-id184304316?k=6&m=184304316&s=612x612&w=0&h=Y3daH7tigcvjNSI7dYKWiT2EWGQ14XHkxdHLqkTpaSU=");
    background-position: center center;
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: 100% 100%;
}

.forecast-header {
    text-align: center;
    font-family: 'Caveat', cursive;
    font-size: 55px;
    color: white;
}

.forecast-container {
    display: flex;
    justify-content: center;
    width: 100%;
    margin: 4% auto;
}

.forecast-card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 160px;
    height: 250px;
    padding: 20px 5px;
}

.forecast-card:not(:last-child){
    border-right: 1px solid white;
}

.date-wrapper, .day-wrapper {
    font-weight: bold;
    font-size: 20px;
}

.date-wrapper {
    font-style: italic;
}

.description-wrapper {
    display: flex;
    flex-grow: 1;
    margin: 20px 0;
    text-wrap: normal;
    text-align: center;
    align-items: center;
}

.temperature-container {
    font-weight: bold;
    font-size: 35px;
}
