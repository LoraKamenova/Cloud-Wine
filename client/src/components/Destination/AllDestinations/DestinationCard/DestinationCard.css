.destination-card {
    display: flex;
    align-items: center;
    background-color: white;
    width: 100%;
    height: 170px ;
    padding: 20px;
    border-radius: 100px;
    margin: 10px;
    box-shadow: 0px 0px 11px -4px #000000;
}

.destination-card-image {
    margin-right: 35px;
    margin-left: 35px;
    /*border-radius: 25%;*/
    width: 130px;
    height: 130px;
}

.destination-card-info-wrapper {
    display: flex;
    flex-direction: column;
}

.destination-info-item {
    font-size: 16px;
}

.destination-card-line-wrapper {
    display: flex;
    margin: 5px 0;
}

.destination-card-pic {
    width: 25px;
    height: 25px;
    margin-right: 10px;
}
