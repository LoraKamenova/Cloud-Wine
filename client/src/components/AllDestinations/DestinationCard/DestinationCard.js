import './DestinationCard.css'

const DestinationCard = (props) => {
    return (
        <article className="destination-card">
            <div className="destination-card-image-wrapper">
                <img className="destination-card-image" src={props.url}/>
            </div>

            <div className="destination-card-info-wrapper">

                <div className="destination-card-line-wrapper">
                    <img className="destination-card-pic" src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1616189586/React%20Project/house_nwweqf.png" />
                    <h3 className="destination-info-item destination-card-title">{props.title}</h3>
                </div>

                <div className="destination-card-line-wrapper">
                    <img className="destination-card-pic" src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1616189587/React%20Project/push-pin_hopc61.png" />
                    <h3 className="destination-info-item destination-card-address">{props.address}</h3>
                </div>

                <div className="destination-card-line-wrapper">
                    <img className="destination-card-pic" src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1616199816/React%20Project/info_rg32io.png" />
                    <h3 className="destination-info-item destination-card-services">{props.services}</h3>
                </div>

            </div>
        </article>
    );
};

export default DestinationCard;
