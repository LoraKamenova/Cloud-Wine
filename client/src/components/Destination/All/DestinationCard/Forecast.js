import './Forecast.css'

import * as destinationService from "../../../../services/destinationService";

import {useEffect, useState} from 'react';

const Forecast = ({match}) => {

    let [weatherInfo, setWeatherInfo] = useState({});

    useEffect(() => {

        destinationService.getWeatherInfo(match.params.lat, match.params.lon)
            .then(res => setWeatherInfo(res));

    }, []);

    console.log(weatherInfo.daily);

    function convertToDate(dateStamp) {
        const milliseconds = dateStamp * 1000
        const dateObject = new Date(milliseconds)
        return dateObject.toLocaleDateString('en-GB',)

    }

    function convertToDay(dateStamp) {
        const milliseconds = dateStamp * 1000
        const dateObject = new Date(milliseconds)

        let day = dateObject.toLocaleDateString('en-GB', {weekday: "long"});

        if (day === "Monday") {
            return "понеделник";
        } else if (day === "Tuesday") {
            return "вторник";
        } else if (day === "Wednesday") {
            return "сряда";
        } else if (day === "Thursday") {
            return "четвъртък";
        } else if (day === "Friday") {
            return "петък";
        } else if (day === "Saturday") {
            return "събота";
        } else if (day === "Sunday") {
            return "неделя";
        }
    }


    return (
        <section className="custom-details-section forecast-section">
            <div className="forecast-wrapper">
                <h2 className="forecast-header">7-дневна прогноза</h2>
                <div className="forecast-container">
                    {weatherInfo.daily.map((x, index) => (index > 0) ? (
                            <div className="forecast-card" key={x._id} {...x}>
                                <div className="date-wrapper">{convertToDate(x.dt)}</div>
                                <div className="day-wrapper">{convertToDay(x.dt)}</div>
                                <div className="description-wrapper">{x.weather[0].description}</div>
                                <div className="temperature-container">
                                    <span>{Math.round(x.temp.min)}&#176;</span>/<span>{Math.round(x.temp.max)}&#176;</span>
                                </div>
                            </div>
                        ) : null
                    )}
                </div>
            </div>
        </section>

    );
};

export default Forecast;
