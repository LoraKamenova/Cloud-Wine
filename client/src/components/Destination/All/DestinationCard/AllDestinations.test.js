import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import DestinationCard from './DestinationCard';

describe('DestinationCard Component', () => {
    it('Should display entry title', () => {
        render(
            <BrowserRouter>
                <DestinationCard title="Шато Копса" link="/"/>
            </BrowserRouter>
        );

        expect(document.querySelector(".destination-card-title").textContent).toBe('Шато Копса');
    });

    it('Should display card address', () => {
        render(
            <BrowserRouter>
                <DestinationCard address="с. Московец, общ. Карлово" link="/"/>
            </BrowserRouter>
        );

        expect(document.querySelector(".destination-card-address").textContent).toBe('с. Московец, общ. Карлово');
    });

    it('Should display card services', () => {
        render(
            <BrowserRouter>
                <DestinationCard services="Дегустация на вина" link="/"/>
            </BrowserRouter>
        );

        expect(document.querySelector(".destination-card-services").textContent).toBe('Дегустация на вина');
    });

    it("DestinationCard Component snapshot", () => {
        const { asFragment } = render(<BrowserRouter><DestinationCard link="/"/></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    });

});
