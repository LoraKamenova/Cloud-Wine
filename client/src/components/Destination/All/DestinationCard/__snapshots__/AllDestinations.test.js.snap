// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationCard Component DestinationCard Component snapshot 1`] = `
<DocumentFragment>
  <article
    class="destination-card"
  >
    <div
      class="destination-card-image-wrapper"
    >
      <img
        alt="logo"
        class="destination-card-image"
      />
    </div>
    <div
      class="destination-card-info-wrapper"
    >
      <div
        class="destination-card-line-wrapper"
      >
        <img
          alt="house"
          class="destination-card-pic"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1616189586/React%20Project/house_nwweqf.png"
        />
        <h3
          class="destination-info-item destination-card-title"
        />
      </div>
      <div
        class="destination-card-line-wrapper"
      >
        <img
          alt="pin"
          class="destination-card-pic"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1616189587/React%20Project/push-pin_hopc61.png"
        />
        <h3
          class="destination-info-item destination-card-address"
        />
      </div>
      <div
        class="destination-card-line-wrapper"
      >
        <img
          alt="info"
          class="destination-card-pic"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1616199816/React%20Project/info_rg32io.png"
        />
        <h3
          class="destination-info-item destination-card-services"
        />
      </div>
    </div>
  </article>
</DocumentFragment>
`;

exports[`RedDictionaryCard Component DestinationCard Component snapshot 1`] = `
<DocumentFragment>
  <article
    class="destination-card"
  >
    <div
      class="destination-card-image-wrapper"
    >
      <img
        alt="logo"
        class="destination-card-image"
      />
    </div>
    <div
      class="destination-card-info-wrapper"
    >
      <div
        class="destination-card-line-wrapper"
      >
        <img
          alt="house"
          class="destination-card-pic"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1616189586/React%20Project/house_nwweqf.png"
        />
        <h3
          class="destination-info-item destination-card-title"
        />
      </div>
      <div
        class="destination-card-line-wrapper"
      >
        <img
          alt="pin"
          class="destination-card-pic"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1616189587/React%20Project/push-pin_hopc61.png"
        />
        <h3
          class="destination-info-item destination-card-address"
        />
      </div>
      <div
        class="destination-card-line-wrapper"
      >
        <img
          alt="info"
          class="destination-card-pic"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1616199816/React%20Project/info_rg32io.png"
        />
        <h3
          class="destination-info-item destination-card-services"
        />
      </div>
    </div>
  </article>
</DocumentFragment>
`;
